File: app/src/main/java/com/example/muzik/response_model/PlayList.java
package com.example.muzik.response_model;


import java.sql.Date;

public class PlayList {
    private long playListID;
    private String type;
    private long userID;
    private String name;
    private Date dateAdded;

    public PlayList(long playListID, String type, long userID, String name, Date dateAdded) {
        this.playListID = playListID;
        this.type = type;
        this.userID = userID;
        this.name = name;
        this.dateAdded = dateAdded;
    }

    public long getPlayListID() {
        return playListID;
    }

    public void setPlayListID(long playListID) {
        this.playListID = playListID;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public long getUserID() {
        return userID;
    }

    public void setUserID(long userID) {
        this.userID = userID;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Date getDateAdded() {
        return dateAdded;
    }

    public void setDateAdded(Date dateAdded) {
        this.dateAdded = dateAdded;
    }
}

File: app/src/main/java/com/example/muzik/data_model/standard_model/Playlist.kt
package com.example.muzik.data_model.standard_model

import android.net.Uri
import com.example.muzik.data_model.retrofit_model.response.PlaylistResponse
import java.util.Date

class Playlist(
    val playlistID: Long? = null,
    val type: String? = null,
    val userID: Long? = -1,
    val userName: String? = null,
    val name: String? = "",
    val dateAdded: Date? = Date(),
    val imageURI: Uri? = Uri.parse("")
) : Model {

    companion object {
        fun buildOnline(playlistResponse: PlaylistResponse): Playlist {
            return Playlist(
                playlistID = playlistResponse.playlistID,
                type = playlistResponse.type,
                userID = playlistResponse.userID,
                name = playlistResponse.name,
                dateAdded = playlistResponse.dateAdded,
                imageURI = Uri.parse(playlistResponse.imageURL)
            )
        }
    }

    fun requirePlaylistID(): Long {
        checkNotNull(playlistID) {
            ("Playlist $this must have non-null playlistID")
        }
        return playlistID
    }
}